runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRLogEntry.java

```diff
@@ -0,0 +1,129 @@
+package org.corfudb.protocols.logprotocol;
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ * A log entry structure which contains a collection of multiSMRentries,
+ * each one contains a list of updates for one object.
+ */
+@SuppressWarnings("checkstyle:abbreviation")
+@ToString
+@Slf4j
+public class SMRLogEntry extends LogEntry {
+
+    // Map from stream-ID to a list of SMR updates to this stream.
+    @Getter
+    public Map<UUID, List<SMRRecord>> entryMap = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Since you are doing this clean up, can you also hide this map. We already expose getStreamEntry, addTo, mergeInto and getSMRUpdates

Reply:
Done. Sometimes we do need to get the updates for all the streams, so I just changed it in another cleaner way: I overloaded `getSMRUpdates(UUID streamID)` with a new API `getSMRUpdates()`, which basically returns updates for all streams.